### PR TITLE
Fix packaging

### PR DESCRIPTION
**Pull request by SchnWalter**
I've updated the setup.py in order to be able to use pip install.

But I've also kept the old `tools/mia` script so that you can run MIA by only installing the docopt and pyyaml python libraries -- without the need for pip or virtualenv.

This required that I move the `templates` folder into the python module -- this is the only way I could make it work properly in all cases... and only after gettings some help on IRC.

### CODE REVIEW

File: mia/__main__.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                based on existing templates.
+    help        Display detailed information about a certain command.
+    install     Install the OS and the built update.zip file onto the device.
+
+See 'mia help <command>' for more information on a specific command.
+
+"""
+
+# Get the path of the mission-impossible-android project.
+import os
```
Comment (patcon) [inline mia/__main__.py]: imports should always be at the top: https://www.python.org/dev/peps/pep-0008/#imports

Comment (SchnWalter) [inline mia/__main__.py]: Done!

**Comment by patcon** [comment]
Hm. getting this error on install:

https://gist.github.com/patcon/c0c0c3715876711c907d

Thoughts?

**Comment by SchnWalter** [comment]
Looks like you tried to install it using Python 2.7. I have several things that don't work yet in 2.7. But i'm going to fix these. For now, instead of running `virtualenv .` you should use `virtualenv -p python3 .` which will use python3 from the system path to build the virtual environment. Or you could use `make prepare` which will also show you a message on how to use virtualenv

https://github.com/patcon/mission-impossible-android/pull/46/files#diff-b67911656ef5d18c4ae36cb6741b7965R32

**Comment by patcon** [comment]
Nice. Thanks. I like the `make prepare` approach. A common entry point is really helpful since so much is changing lately. It might also help use pare down the README instructions that he gotten very long and choose-your-own-adventure-ish :)

And yes, it would be great if we could avoid any reliance on specific pythons, as we shouldn't be doing anything too complicated that would require new features (?)

Figured out the error: PyYaml requires `apt-get install libyaml-dev libpython2.7-dev`

**Comment by SchnWalter** [comment]
We could have a simple usage info in the README... and then point to a wiki page where people can go crazy.

I'll start working on some changes to get the script working in Python 2.7.

**Comment by patcon** [comment]
But don't worry too much about it too much @SchnWalter. I think this is just the friction for me since little things are slowly changing and I'm not staying on top of it -- so long as new people don't hit the friction, it ok :) So yeah, feel free to work on whatever is most important to you

**Comment by patcon** [comment]
mind if `make prepare puts everything into `.venv/` rather than directly in project root?

**Comment by patcon** [comment]
Ah, and I'm not a fan of separate wiki repos like github does -- docs should be version-controlled with code, just like tests, since they apply to a specific point in time

**Comment by patcon** [comment]
I'm about to merge this and I'll make my small changes after.

**Comment by patcon** [comment]
OK, coo, merged in b983e6415eab0d01fdf376c98fed9b24796b2136 (rebased against master)

Thanks @SchnWalter!

**Comment by patcon** [comment]
Hm. Not sure if you want to add your noreply email to your github so that it links them correctly

**Pull request closed without merging**